### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

**Comment by othercorey** [comment]
Not sure why prefer lowest is failing on GA.

**Comment by othercorey** [comment]
> Not sure why prefer lowest is failing on GA.

We have to add plugin-installer dependencies for composer v2 to ensure 1.3+.

**Comment by othercorey** [comment]
@markstory Can you look at the console integration test failures on windows?

File: .travis.yml [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
-  - if [[ $DEFAULT = 1 && $TRAVIS_PHP_VERSION != 7.3 ]]; then vendor/bin/phpunit; fi
-
-  - if [[ $PHPCS = 1 ]]; then vendor/bin/phpcs -n -p src/ tests/; fi
-  - if [[ $STATIC_ANALYSIS = 1 ]]; then composer stan; fi
-
-after_success:
-  - if [[ $DEFAULT = 1 && $TRAVIS_PHP_VERSION = 7.3 ]]; then bash <(curl -s https://codecov.io/bash); fi
-
-notifications:
-  email: false
```
Comment (dereuromark) [inline .travis.yml:50]: Refs https://github.com/cakephp/authentication/pull/419#pullrequestreview-526926081

Comment (othercorey) [reply .travis.yml:50]: The email notification?

Comment (dereuromark) [reply .travis.yml:50]: Wrong line I guess, no, this was referring to the dist XML file to best be kept.

Comment (othercorey) [reply .travis.yml:50]: I didn't change phpunit.xml.dist.

File: .gitattributes [hunk 2]
```diff
@@ -9,9 +9,8 @@
 .gitignore export-ignore
 .github export-ignore
 .scrutinizer.yml export-ignore
-.travis.yml export-ignore
 appveyor.yml export-ignore
 CONTRIBUTING.md export-ignore
 phpunit.xml.dist export-ignore
-phpcs.xml.dist export-ignore
-tests/test_app/ export-ignore
+phpcs.xml export-ignore
```
Comment (dereuromark) [inline .gitattributes:15]: This file
I dont know why my browser jumped to the wrong ones.

File: .travis.yml (hunk 1 above)
Comment (dereuromark) [reply .travis.yml:50]: OK, then all good.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          export CODECOVERAGE=1 && vendor/bin/phpunit --verbose --coverage-clover=coverage.xml
+        else
+          vendor/bin/phpunit
+        fi
+
+    - name: Submit code coverage
+      if: matrix.php-version == '7.4'
+      uses: codecov/codecov-action@v1
+
+  testsuite-windows:
```
Comment (markstory) [inline .github/workflows/ci.yml:94]: Do we need a windows build? I don't think there is anything in this project that is windows specific.

Comment (othercorey) [reply .github/workflows/ci.yml:94]: If we don't need to run the db queries on windows, then no.

Comment (ADmad) [reply .github/workflows/ci.yml:94]: We don't need to test db queries on windows. Let's not contribute to global warming unnecessarily.

Comment (othercorey) [reply .github/workflows/ci.yml:94]: I'm not trying to contribute to global warming here.

Comment (othercorey) [reply .github/workflows/ci.yml]: Removed the windows build

**Comment by othercorey** [comment]
Do we need to test on anything other than sqlite for these?

**Comment by markstory** [comment]
> Do we need to test on anything other than sqlite for these?

I don't think that is necessary right now. We're aren't doing any fancy queries in this plugin.

**Comment by othercorey** [comment]
> > Do we need to test on anything other than sqlite for these?
> 
> I don't think that is necessary right now. We're aren't doing any fancy queries in this plugin.

My point was that we're building on all the dbs, which is why the windows build existed. If we don't need to test all of them this test suite can be smaller.

**Pull request merged**